pkg/broker/handler/pool/pool_test.go

```diff
@@ -81,3 +123,18 @@ func (p *fakeSyncPool) SyncOnce(ctx context.Context) error {
 	}
 	return nil
 }
+
+// GetFreePort asks a free open port.
+func GetFreePort() (int, error) {
+	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
+	if err != nil {
+		return 0, err
+	}
+
+	l, err := net.ListenTCP("tcp", addr)
+	if err != nil {
+		return 0, err
+	}
+	defer l.Close()
```

Comment:
Should let the caller close it, right?

Reply:
This is a trick to get a random port. If we don't close here, the http server won't be able to use this port.

Reply:
Oh I see, you close the listener so the port can be used by the test. I guess in theory there could be a race that something else claims the port as soon as it's released here but it may never happen in reality.